drivers/serial/uart_ns16550.c

```diff
@@ -950,12 +950,21 @@ static int uart_ns16550_line_ctrl_set(const struct device *dev,
 				      uint32_t ctrl, uint32_t val)
 {
 	struct uart_ns16550_dev_data *data = dev->data;
-	uint32_t mdc, chg;
+	const struct uart_ns16550_device_config *const dev_cfg = dev->config;
+	uint32_t mdc, chg, pclk = 0;
 	k_spinlock_key_t key;
 
+	if (dev_cfg->sys_clk_freq != 0U) {
+		pclk = dev_cfg->sys_clk_freq;
+	} else {
+		if (device_is_ready(dev_cfg->clock_dev)) {
```

Comment:
do we need `device_is_ready()` check here?

Reply:
> do we need `device_is_ready()` check here?

It is being checked in configuration api so added it here also.